### PR TITLE
create first user story test for jabbr

### PR DESCRIPTION
**Pull request by sachin303**
updated test to use headless testing utility CasperJs (with phantomjs)

create basic test on user login and registration screen

### CODE REVIEW

File: Build/UserStories.proj [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (davidfowl) [inline Build/UserStories.proj]: Any reason this isn't part of the main build file as another target?

File: JabbR.Tests.UserStories/JabbRUserStoryBase.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
+﻿using JabbR.Models;
+using JabbR.Services;
+
+namespace JabbR.Tests.UserStories {
+    
+    public class JabbRUserStoryHelper
```
Comment (davidfowl) [inline JabbR.Tests.UserStories/JabbRUserStoryBase.cs]: Fix the formatting in this file.

File: JabbR.Tests.UserStories/JabbRLoginTest.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Firefox;
+using Xunit;
+
+namespace JabbR.Tests.UserStories
+{
+    public class JabbRLoginTest : IDisposable
+    {
+        private OpenQA.Selenium.Firefox.FirefoxDriver driver;
```
Comment (davidfowl) [inline JabbR.Tests.UserStories/JabbRLoginTest.cs]: Don't use the full namespace.

File: JabbR.Tests.UserStories/JabbRLoginTest.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        private JabbRUserStoryHelper helper = new JabbRUserStoryHelper();
```
Comment (davidfowl) [inline JabbR.Tests.UserStories/JabbRLoginTest.cs]: Private variables start with _

Comment (sachin303) [inline JabbR.Tests.UserStories/JabbRLoginTest.cs]: ooh, I will fix it in all my changes

File: JabbR.Tests.UserStories/JabbRLoginTest.cs (hunk 3 above)
Comment (sachin303) [inline JabbR.Tests.UserStories/JabbRLoginTest.cs]: yup, Lesson no: 2, Will fix it

File: Build/UserStories.proj (hunk 1 above)
Comment (sachin303) [inline Build/UserStories.proj]: Yeah, user stories need deployed jabbr app in local IIS/IIS express. So, It should not be part of build & Test.

What you say about that?

Comment (sachin303) [inline Build/UserStories.proj]: I thought to write PS script to deploy jabbr in local IIS and then run user stories. What you say ?

Comment (davidfowl) [inline Build/UserStories.proj]: Nope

Comment (davidfowl) [inline Build/UserStories.proj]: Just make it a new target

**Comment by sachin303** [comment]
Fix the formatting
change private variable declaration
merge new target in single build

File: Build/Build.proj [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="4.0" DefaultTargets="RunAll" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+<Project ToolsVersion="4.0" DefaultTargets="$(Targets)" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (davidfowl) [inline Build/Build.proj]: Um no. Change that back to RunAll.

File: Build/Build.proj [hunk 6]
```diff
@@ -11,6 +11,7 @@
     
     <ItemGroup>
         <TestProjects Include="$(ProjectRoot)\JabbR.Tests\bin\$(Configuration)\JabbR.Tests.dll" />
+		<UserStoriesProjects Include="$(ProjectRoot)\JabbR.Tests.UserStories\bin\$(Configuration)\JabbR.Tests.UserStories.dll" />
```
Comment (davidfowl) [inline Build/Build.proj]: Fix the formatting.

File: Build/Build.proj [hunk 7]
```diff
@@ -25,7 +26,11 @@
     <Target Name="RunTests">
         <xunit Assembly="%(TestProjects.FullPath)" />
     </Target>
-    
+	
+	<Target Name="RunUserStoriesTests">
```
Comment (davidfowl) [inline Build/Build.proj]: Fix the formatting.

File: JabbR.Tests.UserStories/JabbRLoginTest.cs [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+{
+    public class JabbRLoginTest : IDisposable
+    {
+        private FirefoxDriver _driver;
+        
+        private const string _jabbrBaseUrl = "http://localhost:16207/";
+        private const string _testuser1 = "testuser1";
+        private const string _testuser1pwd = "testuser1";
+        
+        //private const string testuser1email = "abc@xyz.com";
```
Comment (davidfowl) [inline JabbR.Tests.UserStories/JabbRLoginTest.cs]: Why are these commented out?

File: JabbR.Tests.UserStories/JabbRLoginTest.cs [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+        //private JabbRUserStoryHelper _helper = new JabbRUserStoryHelper( );
```
Comment (davidfowl) [inline JabbR.Tests.UserStories/JabbRLoginTest.cs]: Why is this commented out?

File: JabbR.Tests.UserStories/JabbRUserStoryBase.cs [hunk 10]
```diff
@@ -0,0 +1,31 @@
+﻿using JabbR.Models;
+using JabbR.Services;
+
+namespace JabbR.Tests.UserStories 
+{   
+    public class JabbRUserStoryHelper
+    {
+        private   IMembershipService _membershipService;
```
Comment (davidfowl) [inline JabbR.Tests.UserStories/JabbRUserStoryBase.cs]: Fix the formatting in this entire file.

File: Build/Build.proj (hunk 6 above)
Comment (sachin303) [inline Build/Build.proj]: fixed it

File: Build/Build.proj (hunk 7 above)
Comment (sachin303) [inline Build/Build.proj]: fixed

File: JabbR.Tests.UserStories/JabbRUserStoryBase.cs [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+        private ICryptoService _criptoService;
+
+        public JabbRUserStoryHelper ( )
+        {
+            Setup();
+        }
+
+        public void Setup()
+        {
+            _repository = new PersistedRepository(new JabbrContext());
```
Comment (samandmoore) [inline JabbR.Tests.UserStories/JabbRUserStoryBase.cs]: seems like we should use the `InMemoryRepository`, no? we probably don't want to have a db dependency in these tests.

**Comment by samandmoore** [comment]
so, i pulled this down locally and tried to run it and it failed because it couldn't find the browser driver. does that need to be included in the repo and path referenced?

**Comment by samandmoore** [comment]
does this run headless or does it open a browser? we'd prefer headless

File: JabbR.Tests.UserStories/JabbRUserStoryBase.cs (hunk 11 above)
Comment (sachin303) [inline JabbR.Tests.UserStories/JabbRUserStoryBase.cs]: Yes, I agree. I will use InMemoryRepo. Currently, we only have one test case which do not use it.

**Comment by sachin303** [comment]
You need to have JabbR website running (localhost:16207). Browser drive runs on this URL.

**Pull request closed without merging**